### PR TITLE
Bug/Class Label Issue

### PR DESCRIPTION
**Pull request by justinbt1**
## Class Labelling Issues
Proposed changes that seek to resolve the label mismatch between the `value_counts` JSON file used as class labels in `train.py` and the actual order of classes in the parquet dataset. See issue https://github.com/wellcometrust/grant_hrcs_tagger/issues/10 for full investigation.

### Changes Made
The label mismatch issue is avoided by using the train and test dataframe column headers as class labels instead of the separate `value_counts.json` file, as these are already correctly ordered in the same lexicographical order as the data. Value counts are then explicitly calculated for both the train and test dataset in `test.py` using `np.sum()`. 

As value_counts code is no longer needed, code to generate, transform and use it has been removed from `preprocess.py` and `train.py`. This also has the added benefit of significantly reducing code complexity, and improving the robustness of the codebase.

To prevent leakage of assumption on underlying class distribution between training and test, now only the count of class occurrences in the training data is passed to the `WeightedTrainner` class. Alongside this the logic for calculating the class_weights tensor has been moved to inside the `if class_weighting:` block, so it is now only calculated if needed.

Modified W&B metrics logging to now log the class occurrence counts separately for the train, test datasets, instead of the sum of both as previously logged using `value_counts`. This will allow improved understanding of the size of each class in each dataset and make it easier to validate the train, test strategy used.

### Testing These Changes
Tested labelling is consistent and correct in all intermediate outputs and now appears in the expected lexicographic order in the W&B logs.

### Evaluation of Change Impact
To compare training logs in W&B with the original and new labels, a new run was performed using the same parameters as the old run. (30 epochs, batch size of 16, output weighting True).

These can be viewed in W&B:
- New: https://wandb.wellcome-data.org/wellcome-ml-team/grant_hrcs_tagger/runs/pe6li3ls
- Old: https://wandb.wellcome-data.org/wellcome-ml-team/grant_hrcs_tagger/runs/hvum3szd

Unfortunately comparing the value counts against the class labels shows no change, as these are correctly matched. There is however a statistically significant difference between the metric distribution between the two groups.

#### RA codes plotted against F1 for the original labelling:
output
The above plot shows only a very weak correlation that is not statistically significant between occurrence of training data for a class and the F1 score. Meaning we can't reject the null hypothesis that the distribution of F1 scores is random, or that there is no correlation between the two.

R Coefficient: 0.18177960860695436
P Value: 0.21625730815412547

#### RA codes plotted against F1 for the new labelling:
outputNew
With the new labelling there is a statistically significant, moderate correlation between value count and F1 performance, which is more inline with expected behaviour. This suggests the issue for poor macro-F1 performance is the lack of training data for certain classes.

R Coefficient: 0.4326715660968124
P Value: 0.0021306723207720007

As expected no significant change in model performance was noted.

### CODE REVIEW

**Comment by ArneRobben** [comment]
This repo resolves https://github.com/wellcometrust/grant_hrcs_tagger/issues/10

Proposed changes also clean up the codebase and avoid the use of the `value_counts.json`
